apps/web-client/src/lib/db/cr-sqlite/history.ts

```diff
@@ -27,13 +60,35 @@ export function getPastNotes(db: DB, date: string): Promise<PastNoteItem[]> {
 	return db.execO(query, [date]);
 }
 
+/**
+ * Parameters for filtering transaction history queries
+ */
 type Params = {
 	isbn?: string;
 	warehouseId?: number;
 	startDate?: Date;
 	endDate?: Date;
 };
 
+/**
+ * Retrieves historical transactions with flexible filtering options.
+ * Can filter by ISBN, warehouse, and date range.
+ * Returns detailed information about each transaction including book metadata.
+ * Only includes transactions from committed notes.
+ *
+ * @param {DB} db - Database connection
+ * @param {Params} params - Query filters:
+ *   - isbn: Filter by specific book
+ *   - warehouseId: Filter by specific warehouse
+ *   - startDate: Include transactions after this date
+ *   - endDate: Include transactions before this date
+ * @returns {Promise<PastTransactionItem[]>} Historical transactions with:
+ *   - Basic info (isbn, title, author, quantity)
+ *   - Pricing (price, discount)
+ *   - Location (warehouseId, warehouseName)
+ *   - Note details (noteId, noteName, noteType)
+ *   - Timestamp (committedAt)
```

Comment:
As usual, don't specify what the types contain.

```suggestion
 * @returns {Promise<PastTransactionItem[]>} Historical transactions
```